plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemFirework.java

```diff
@@ -75,12 +74,42 @@ public ListTag getFireworkData() {
         return list;
     }
 
-    @Override
-    public ObjectTag getObjectAttribute(Attribute attribute) {
-
-        if (attribute == null) {
-            return null;
+    public MapTag getFireworkDataMap(){
+        List<FireworkEffect> effects;
+        MapTag map = new MapTag();
+        if (item.getItemMeta() instanceof FireworkMeta) {
+            FireworkMeta meta = (FireworkMeta) item.getItemMeta();
+            effects = meta.getEffects();
+            int power = meta.getPower();
+            if (power != 0) {
+                map.putObject("power", new ElementTag(power));
+            }
         }
+        else {
+            effects = Collections.singletonList(((FireworkEffectMeta) item.getItemMeta()).getEffect());
+        }
+        if (effects != null) {
+            int size = effects.size();
+            for (int i = 0; i < size; i++) {
+                FireworkEffect effect = effects.get(i);
+                if (effect == null) {
+                    continue;
+                }
+                Color ColOne = effect.getColors() != null && effect.getColors().size() > 0 ? effect.getColors().get(0) : Color.BLUE;
+                Color ColTwo = effect.getFadeColors() != null && effect.getFadeColors().size() > 0 ? effect.getFadeColors().get(0) : ColOne;
+                MapTag effectMap = new MapTag();
+                effectMap.putObject("trail", new ElementTag(effect.hasTrail()));
+                effectMap.putObject("flicker", new ElementTag(effect.hasFlicker()));
+                effectMap.putObject("type", new ElementTag(effect.getType().name()));
+                effectMap.putObject("color", new ColorTag(ColOne));
+                effectMap.putObject("fadeColor", new ColorTag(ColTwo));
+                map.putObject(String.valueOf(i+1), effectMap);
```

Comment:
This is a list. You've made a list using MapTags. Use a list.

Reply:
Just thought that format would be easier to use - I.e.
```yaml
  - definemap Firework:
      power: 3
      1:
        color: red
        fadeColor: white
        trail: true
      2:
        color: green
        type: star
        flicker: true
```

Reply:
A tag is not a script. A tag is a tag.